db/db_impl/db_impl.h

```diff
@@ -162,15 +162,25 @@ class Directories {
   std::unique_ptr<FSDirectory> wal_dir_;
 };
 
-struct DBOpenLogReporter : public log::Reader::Reporter {
+struct DBOpenLogRecordReadReporter : public log::Reader::Reporter {
```

Comment:
Rename to better indicate what level of corruption it reports. The level may change in the future once we figure out how to consolidate all the error status in ProcessLogFile() (see existing TODOs)

Reply:
nit: alternatively, we can have a `Reporter` initialized with the `wal_number_` it is opened for. If a `Reporter::Corruption(, corrupted_wal_number=kMaxSequence)` call is not providing a specified wal number, it will track its own `wal_number_` as the `corrupted_wal_number_`. I think that can help eliminate the need of a dedicated `PredecessorLogCorruption` API. But it's up to you, not a big deal.

This may help us with removing the if else in `HandleNonOkStatusOrOldLogRecord`, but I agree with you that before we can consolidate the status handling, it's safer to keep it like this.

Reply:
I see let me think more and fix it!

Reply:
Fixed excluding the constructor part, which I found is better done with the status consolidation.